src/makielayout/blocks/polaraxis.jl

```diff
@@ -0,0 +1,537 @@
+        @show(mins, maxs)
+        return Rect2f(mins,maxs)
+    end
+    for x in range(xmin, xmax; length = N)
+        for y in range(ymin, ymax; length = N)
+            u, v = Makie.apply_transform(f, Point(x, y))
+            umin = min(umin, u)
+            umax = max(umax, u)
+            vmin = min(vmin, v)
+            vmax = max(vmax, v)
+        end
+    end
+
+    return Rect(Vec2(umin, vmin), Vec2(umax-umin, vmax-vmin))
+end
+
+
+# Define its inverse (for interactivity)
+Makie.inverse_transform(trans::PolarAxisTransformation) = Makie.PointTrans{2}() do point
+    Point2f(hypot(point[1], point[2]), -trans.direction * (atan(point[2], point[1]) - trans.θ_0))
+end
+
+# End transform code
+
+# Some useful code to transform from data (transformed) space to pixelspace
+
+function project_to_pixelspace(scene, point::VecTypes{N, T}) where {N, T}
+    @assert N ≤ 3
+    return to_ndim(
+        typeof(point),
+        Makie.project(
+            # obtain the camera of the Scene which will project to its screenspace
+            camera(scene),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            :data, :pixel,
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                point
+            )
+        ),
+        0.0
+    )
+end
+
+function project_to_pixelspace(scene, points::AbstractVector{Point{N, T}}) where {N, T}
+    to_ndim.(
+        Ref(eltype(points)),
+        Makie.project.(
+            # obtain the camera of the Scene which will project to its screenspace
+            Ref(Makie.camera(scene)),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            Ref(:data), Ref(:pixel),
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                points
+            )
+        ),
+        Ref(0.0)
+    )
+end
+
+# A function which redoes text layouting, to provide a bbox for arbitrary text.
+
+function text_bbox(textstring::AbstractString, textsize::Union{AbstractVector, Number}, font, align, rotation, justification, lineheight, word_wrap_width = -1)
+    glyph_collection = Makie.layout_text(
+            textstring, textsize,
+            font, align, rotation, justification, lineheight,
+            RGBAf(0,0,0,0), RGBAf(0,0,0,0), 0f0, word_wrap_width
+        )
+
+    return Rect2f(Makie.boundingbox(glyph_collection, Point3f(0), Makie.to_rotation(rotation)))
+end
+
+function text_bbox(plot::Text)
+    return text_bbox(
+        plot.text[],
+        plot.textsize[],
+        plot.font[],
+        plot.align[],
+        plot.rotation[],
+        plot.justification[],
+        plot.lineheight[],
+        RGBAf(0,0,0,0), RGBAf(0,0,0,0), 0f0,
+        plot.word_wrap_width[]
+    )
+end
```

Comment:
It's probably better to plot text first, and then use normal bounding boxes to update things. Otherwise changes and optimizations there might get left behind here. (I think right now it's more or less the same amount of work. plot first just needs to guard against infinite loops.)

Reply:
I'll definitely look further into the structure of text then.  I was under the impression that everything got squashed into one GlyphCollection but could definitely be wrong.

Reply:
Also, if the GlyphCollection changes when position changes, then it's harder to lift off it, since I only care about the text extents of each segment - not their positions (at least for now).  If we want a 100% tight bbox then it can be tried, but I would definitely like to do that later.

Reply:
Oh, you're doing bboxes per string in a vector with that. If you're doing `boundbox(p)` they'll get combined, but the glyph collections should be available per string.

 > GlyphCollection changes when position changes

glyph collections shouldn't trigger on `plot.position` anymore. If they are we should clean that up, because there is no reason to mix them until the final rendering code.